### PR TITLE
Shortfin llm `temperature` + `SamplingParams` dataclass

### PR DESCRIPTION
**Pull request by stbaione**
# Description

Implement `temperature` in the `Shortfin LLM Server`.

Great explanation for temperature from [llama.cpp docs](https://github.com/ggml-org/llama.cpp/blob/master/examples/main/README.md?#temperature).

TLDR:
- `temperature < 1.0` => more deterministic output
- `temperature > 1.0` => less deterministic output

*There won't be an impact on the output tokens generated, until I add `nucleas sampling`. The max stays the max, regardless of if the logits were divided by a scalar.*

I was a little confused on what the use for the temperature range. I found some articles that said the "official" llama temperature range is between [0, 1]. But, the linked `llama.cpp` doc above mentions a temperature above `1.0`.

The OpenAI API allows temperatures in the range of [0.0, 2.0], so I just went with that.

There may be a better choice of `min_temperature` as well. Lemme know what you think...

# Changes
- `apply_temperature` function added to base `Beam` class
- `MultiGreedyBeam` is now `GreedyBeam`. Both `Greedy` and `MultiGreedy` now use this class. This causes less repetitive code, and makes the implementations more similar. Greedy was starting to feel unique in some ways.
- `temperature` is stored in `DecodeConfig`. `max_completion_tokens` also is pushed back to `DecodeConfig` for consistency.
- Added `temperature` to the `cli` module
- Added a call to `configure_main_logger` in `cli` module to fix logging bug

### CODE REVIEW

**Review by rsuderman — APPROVED** [review]

File: shortfin/tests/apps/llm/components/token_selection_strategy/greedy_token_selection_strategy_test.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 import logging
 import math
+from typing import Any
```
[the remark this answers is not in the document]
Comment (stbaione) [reply shortfin/tests/apps/llm/components/token_selection_strategy/greedy_token_selection_strategy_test.py]: Oops, thanks for finding this. Updated
